### ISSUE TITLE
Usage of kglm-model

### ISSUE THREAD

**Issue by almoslmi** [rloganiv/kglm-model#46]
I would like to thank you @rloganiv for such great work. I am quite new in the area and intend to apply your model for my data. I did implement the model successfully but didn't manage to apply it for my data. 
Is there any guideline showing how to start or how I can use your model on my data? How can your pre-trained model be used for my data? I would appreciate if you can provide an example of usage on any sentence/text.

Thank you

**Comment by rloganiv** [comment]
I will need more details about what you are trying to do in order to help. Are you trying to evaluate perplexity of the model on your data? Or are you trying to perform additional sentence completion experiments?

**Comment by almoslmi** [comment]
Thanks dear @rloganiv  for your kind reply. Here is what I intend to do.

I intend to use the model to create my graph as it should be empty at first according to the paper. Then want to use sentence completion to help me complete my sentence/paragraph. After that, if possible, I want to get the entity linking to the KB like Wikidata so I can get exactly something similar to Figure 1 in your paper (a list of all entities in my paragraph+ their linking number to Wikidata). 

I have implemented the model as I mentioned previously but didn't know what I should do first and next in order to do that.

**Comment by rloganiv** [comment]
Are the sentence fragments you want the model to complete annotated or unannotated?

If the sentence fragments are unannotated then what you are proposing is rather difficult. Since the KGLM is a forward generative model it is not really well-suited to annotating an existing piece of text. To give an example, in the sentence "I am excited about Tool 's new album", the KGLM would have to predict the entity label [Q184827](https://wikidata.org/wiki/Q184827) before seeing the token "Tool".

**Comment by almoslmi** [comment]
Thanks dear @rloganiv for your reply again

It is unannotated. Simply I intend to apply your model in a domain where the system should be able to help the writer to suggest a sentence completion (based on local KG) for his writing and bring him a list of all entities with their linking to KB after he finishes writing. Hopefully, this will works.

**Comment by rloganiv** [comment]
Hi @almoslmi,

Sorry about the delayed response. Unfortunately the repo is not currently set up to support your desired application. The main difficulty is having the system generate annotations - the discriminative model used for importance sampling is currently not accurate enough to greedily decode an annotation.

You could consider using our annotation code at: https://github.com/rloganiv/kglm-data, however this code is specifically designed for annotating Wikipedia articles and is not well suited for an online setting.
